src/Compilers/CSharp/Test/Emit3/PartialEventsAndConstructorsTests.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests;
+
+public sealed class PartialEventsAndConstructorsTests : CSharpTestBase
+{
+    [Fact]
+    public void ReturningPartialType_LocalFunction_InMethod()
+    {
+        var source = """
+            class @partial
+            {
+                static void Main()
+                {
+                    System.Console.Write(F().GetType().Name);
+                    partial F() => new();
+                }
+            }
+            """;
+        CompileAndVerify(source, parseOptions: TestOptions.Regular13, expectedOutput: "partial").VerifyDiagnostics();
+
+        var expectedDiagnostics = new[]
+        {
+            // (5,30): error CS0103: The name 'F' does not exist in the current context
+            //         System.Console.Write(F().GetType().Name);
+            Diagnostic(ErrorCode.ERR_NameNotInContext, "F").WithArguments("F").WithLocation(5, 30),
+            // (5,50): error CS1513: } expected
+            //         System.Console.Write(F().GetType().Name);
+            Diagnostic(ErrorCode.ERR_RbraceExpected, "").WithLocation(5, 50),
+            // (6,9): error CS0267: The 'partial' modifier can only appear immediately before 'class', 'record', 'struct', 'interface', or a method or property return type.
```

Comment:
>or a method or property return type [](http://example.com/codeflow?start=136&length=35)

Should this error message be updated to include event or constructor?

Reply:
Yes, it will be in the next PR (I already have that locally so I would skip adding a prototype comment for that if that's okay)